### PR TITLE
Add new_test/test_parallel_master_taskloop_simd_device.F90

### PR DESCRIPTION
**Pull request by seyonglee**
Fails with both XLF 16.1.1-10 and GCC 11.1.0
Update test_parallel_master_taskloop_simd_device.c too.

### CODE REVIEW

File: tests/5.0/parallel_master_taskloop_simd/test_parallel_master_taskloop_simd_device.F90 [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+       x(i) = x(i) + y(i) * z(i)
+    END DO
+    !$omp end parallel master taskloop simd
+    !$omp end target
+
+    DO i = 1, N
+       OMPVV_TEST_AND_SET_VERBOSE(errors, x(i) .ne. 1 + y(i)*z(i))
+    END DO
+
+    OMPVV_INFOMSG("This test does not guarantee thread parallelism of the clause.")
```
Comment (spophale) [inline tests/5.0/parallel_master_taskloop_simd/test_parallel_master_taskloop_simd_device.F90:53]: Maybe say "This thread does not guarantee vector instructions were generated for the simd construct".

File: tests/5.0/parallel_master_taskloop_simd/test_parallel_master_taskloop_simd_device.F90 [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+       y(i) = i + 1
+       z(i) = 2*(i + 1)
+    END DO
+
+    !$omp target map(tofrom: x) map(to: y, z)
+    !$omp parallel master taskloop simd num_threads(OMPVV_NUM_THREADS_DEVICE) shared(x, y, z) 
+    DO i = 1, N
+       x(i) = x(i) + y(i) * z(i)
+    END DO
+    !$omp end parallel master taskloop simd
```
Comment (spophale) [inline tests/5.0/parallel_master_taskloop_simd/test_parallel_master_taskloop_simd_device.F90:46]: Comparing to the other tests, the num_threads check has been removed. Maybe we need to add it back for this and C tests to ensure that there is parallelism.

File: tests/5.0/parallel_master_taskloop_simd/test_parallel_master_taskloop_simd_device.F90 (hunk 1 above)
Comment (seyonglee) [reply tests/5.0/parallel_master_taskloop_simd/test_parallel_master_taskloop_simd_device.F90]: Updated both C and Fortran versions.

File: tests/5.0/parallel_master_taskloop_simd/test_parallel_master_taskloop_simd_device.F90 (hunk 2 above)
Comment (seyonglee) [reply tests/5.0/parallel_master_taskloop_simd/test_parallel_master_taskloop_simd_device.F90:50]: Modified both C and Fortran versions to check the number of threads in a parallel region.

**Review by tmh97 — APPROVED** [review]
looks-good-to-go

**Pull request merged**